src/Pim/Bundle/CatalogBundle/EventSubscriber/IndexProductsSubscriber.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\EventSubscriber;
+
+use Akeneo\Bundle\ElasticsearchBundle\Client;
+use Akeneo\Component\StorageUtils\StorageEvents;
+use Pim\Component\Catalog\Model\ProductInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Index products in the search engine.
+ *
+ * This is not done directly in the product saver as it's only a technical
+ * problem. The product saver only handles business stuff.
+ *
+ * @author    Julien Janvier <j.janvier@gmail.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class IndexProductsSubscriber implements EventSubscriberInterface
+{
+    /** @var NormalizerInterface */
+    protected $normalizer;
+
+    /** @var Client */
+    protected $indexer;
+
+    /** @var string */
+    protected $indexType;
+
+    /**
+     * @param NormalizerInterface $normalizer
+     * @param Client              $indexer
+     * @param string              $indexType
+     */
+    public function __construct(NormalizerInterface $normalizer, Client $indexer, $indexType)
+    {
+        $this->normalizer = $normalizer;
+        $this->indexer = $indexer;
+        $this->indexType = $indexType;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            StorageEvents::POST_SAVE => 'indexProduct',
+            StorageEvents::POST_SAVE_ALL => 'bulkIndexProducts',
+        ];
+    }
+
+    /**
+     * Index one single product.
+     *
+     * @param GenericEvent $event
+     */
+    public function indexProduct(GenericEvent $event)
+    {
+        $product = $event->getSubject();
+        if (!$product instanceof ProductInterface) {
+            return;
+        }
+
+        if (!$event->hasArgument('unitary') || false === $event->getArgument('unitary')) {
+            return;
+        }
+
+        $normalizedProduct = $this->normalizer->normalize($product, 'indexing');
```

Comment:
It doesn't make sense that the subscriber is the responsible for normalizing and then indexing an array.

If I want to index my products from another point in the application (a full reindex command will be necessary for example), I don't want to use this subscriber...

Also, normalizing to an array is a specificity of the Elasticsearch indexer. it's internal mechanic. Another search engine could work with something else than array. So it's the ES indexer internal logic, it shouldn't be exposed in a generic product subscriber whose aim is to make sure product are indexed.

So the index function must take product as parameter, as, again, the normalization is already an implementation detail.